### ISSUE TITLE
Failure to respect accessibility in atomic expression

### ISSUE THREAD

**Issue by CaseyCarter** [saarraz/clang-concepts-old#51]
AFAICS, this program fragment:
```c++
class S {
  constexpr bool f() { return true; }
};

template<class T> concept C = T{}.f();

static_assert(!C<S>);
```
should compile successfully. `S::f` is inaccessible, so `T{}.f()` is an invalid expression after substitution of `S` for `T`.

**Comment by saarraz** [comment]
Fixed in latest commit.
